tests/unit/r2i/tensorrt/engine.cc

```diff
@@ -0,0 +1,163 @@
+/* Copyright (C) 2018-2020 RidgeRun, LLC (http://www.ridgerun.com)
+ * All Rights Reserved.
+ *
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+ */
+
+#include <r2i/r2i.h>
+#include <r2i/tensorrt/engine.h>
+#include <r2i/tensorrt/model.h>
+#include <r2i/tensorrt/frame.h>
+#include <r2i/tensorrt/prediction.h>
+
+#include <CppUTest/CommandLineTestRunner.h>
+#include <CppUTest/MemoryLeakDetectorNewMacros.h>
+#include <CppUTest/MemoryLeakDetectorMallocMacros.h>
+#include <CppUTest/TestHarness.h>
+
+#include "mockcudaengine.cc"
+
+class MockModel : public r2i::IModel {
+  r2i::RuntimeError Start (const std::string &name) override {
+    r2i::RuntimeError error; return error;
+  }
+};
+
+void IExecutionContextDeleter (nvinfer1::IExecutionContext *p) {
+  if (p)
+    p->destroy ();
+}
+
+static void ICudaEngineDeleter (nvinfer1::ICudaEngine *p) {
+  if (p)
+    p->destroy ();
+}
+
+TEST_GROUP (TensorRTEngine) {
+  r2i::tensorrt::Engine engine;
+  std::shared_ptr<r2i::tensorrt::Model> model;
+  std::shared_ptr<r2i::IModel> inc_model;
+  std::shared_ptr<r2i::IFrame> frame;
+  r2i::RuntimeError error;
+
+  std::shared_ptr<nvinfer1::IExecutionContext> context;
+  std::shared_ptr<nvinfer1::ICudaEngine> cuda_engine;
+
+  void setup () {
+    error.Clean();
+
+    context = std::shared_ptr<nvinfer1::IExecutionContext> (new
+              nvinfer1::MockExecutionContext,
+              IExecutionContextDeleter);
+
+    cuda_engine = std::shared_ptr<nvinfer1::ICudaEngine> (new
+                  nvinfer1::MockCudaEngine,
+                  ICudaEngineDeleter);
+
+    model = std::make_shared<r2i::tensorrt::Model> ();
+    model->SetContext (context);
+    model->SetCudaEngine (cuda_engine);
+
+    inc_model = std::make_shared<MockModel> ();
+    frame = std::make_shared<r2i::tensorrt::Frame> ();
+  }
+};
+
+TEST (TensorRTEngine, SetModel) {
+  error = engine.SetModel (model);
+  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
+}
+
+TEST (TensorRTEngine, SetModelNull) {
+  error = engine.SetModel (nullptr);
+  LONGS_EQUAL (r2i::RuntimeError::Code::NULL_PARAMETER, error.GetCode ());
+}
+TEST (TensorRTEngine, SetModelInvalid) {
+  error = engine.SetModel (inc_model);
+  LONGS_EQUAL (r2i::RuntimeError::Code::INCOMPATIBLE_MODEL, error.GetCode ());
+}
+
+TEST (TensorRTEngine, StartEngine) {
```

Comment:
Failed start?

Reply:
This engine doesn't have a start logic, this test is only for the interface implemetation, see [here](https://github.com/RidgeRun/r2inference/blob/cc4c5646924b4706c60cc0159275aa61d643b393/r2i/tensorrt/engine.cc#L67)